dask/array/core.py

```diff
@@ -1567,6 +1568,12 @@ def handle_nonmatching_names(func, args, kwargs):
         da_func = getattr(module, func.__name__)
         if da_func is func:
             return handle_nonmatching_names(func, args, kwargs)
+
+        # If ``like`` is contained in ``da_func``'s signature, add ``like=self``
+        # to the kwargs dictionary.
+        if "like" in inspect.getfullargspec(da_func).kwonlyargs:
```

Comment:
Can probably replace with something like `if has_keyword("like", da_func):`

https://github.com/dask/dask/blob/bb0a2d2f56b0e12ce0fa23a8aaad1203f8da36f7/dask/utils.py#L1557

Reply:
Good idea, that doesn't check specifically for a keyword-only argument but I believe it should be no problem for the Dask case. Done in https://github.com/dask/dask/pull/8054/commits/9d7b492773e1a3843c26cce694ffda3c9658455b .